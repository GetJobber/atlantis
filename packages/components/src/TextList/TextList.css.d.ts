declare const styles: {
  readonly "textList": string;
  readonly "bulleted": string;
  readonly "numbered": string;
  readonly "textListItem": string;
};
export = styles;

