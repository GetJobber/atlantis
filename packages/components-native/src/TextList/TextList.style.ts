import { StyleSheet } from "react-native";
import { tokens } from "../utils/design";

export const styles = StyleSheet.create({
  details: {
    width: "100%",
    flexDirection: "column",
  },
  detail: {
    flexDirection: "row",
    paddingLeft: tokens["space-small"],
  },
  detailText: {
    paddingLeft: tokens["space-small"],
    flex: 1,
  },
});
