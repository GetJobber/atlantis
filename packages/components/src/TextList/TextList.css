.textList {
  margin: 0;
  padding: 0;
  padding-left: var(--space-base);
  list-style: disc outside none;
}

.numbered {
  list-style-type: decimal;
}

.textListItem {
  display: list-item;
}
