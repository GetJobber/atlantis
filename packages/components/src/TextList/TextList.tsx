import React, { Children, ReactElement } from "react";
import { TextListItem } from "./TextListItem";

interface TextListProps {
  readonly data: Array<string | ReactElement>;
  readonly type?: "bullets" | "numbers";
}

export function TextList({ data, type = "bullets" }: TextListProps) {
  const Tag = type === "bullets" ? "ul" : "ol";

  return (
    <Tag>
      {Children.map(data, listItem => (
        <TextListItem>{listItem}</TextListItem>
      ))}
    </Tag>
  );
}
