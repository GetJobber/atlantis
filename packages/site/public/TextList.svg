<svg width="224" height="117" viewBox="0 0 224 117" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M53 19C53 16.7909 54.7909 15 57 15H167C169.209 15 171 16.7909 171 19V99C171 101.209 169.209 103 167 103H57C54.7909 103 53 101.209 53 99V19Z" fill="white"/>
<path d="M73 38.8636C72.5417 38.8636 72.1231 38.7519 71.7443 38.5284C71.3655 38.3011 71.0625 37.9981 70.8352 37.6193C70.6117 37.2405 70.5 36.822 70.5 36.3636C70.5 35.9015 70.6117 35.483 70.8352 35.108C71.0625 34.7292 71.3655 34.428 71.7443 34.2045C72.1231 33.9773 72.5417 33.8636 73 33.8636C73.4621 33.8636 73.8807 33.9773 74.2557 34.2045C74.6345 34.428 74.9356 34.7292 75.1591 35.108C75.3864 35.483 75.5 35.9015 75.5 36.3636C75.5 36.822 75.3864 37.2405 75.1591 37.6193C74.9356 37.9981 74.6345 38.3011 74.2557 38.5284C73.8807 38.7519 73.4621 38.8636 73 38.8636ZM85.7727 30.6136V29.3636H94.5V30.6136H90.8409V41H89.4318V30.6136H85.7727ZM98.8864 41.1818C98.0455 41.1818 97.3201 40.9962 96.7102 40.625C96.1042 40.25 95.6364 39.7273 95.3068 39.0568C94.9811 38.3826 94.8182 37.5985 94.8182 36.7045C94.8182 35.8106 94.9811 35.0227 95.3068 34.3409C95.6364 33.6553 96.0947 33.1212 96.6818 32.7386C97.2727 32.3523 97.9621 32.1591 98.75 32.1591C99.2045 32.1591 99.6534 32.2348 100.097 32.3864C100.54 32.5379 100.943 32.7841 101.307 33.125C101.67 33.4621 101.96 33.9091 102.176 34.4659C102.392 35.0227 102.5 35.7083 102.5 36.5227V37.0909H95.7727V35.9318H101.136C101.136 35.4394 101.038 35 100.841 34.6136C100.648 34.2273 100.371 33.9223 100.011 33.6989C99.6553 33.4754 99.2348 33.3636 98.75 33.3636C98.2159 33.3636 97.7538 33.4962 97.3636 33.7614C96.9773 34.0227 96.6799 34.3636 96.4716 34.7841C96.2633 35.2045 96.1591 35.6553 96.1591 36.1364V36.9091C96.1591 37.5682 96.2727 38.1269 96.5 38.5852C96.7311 39.0398 97.0511 39.3864 97.4602 39.625C97.8693 39.8598 98.3447 39.9773 98.8864 39.9773C99.2386 39.9773 99.5568 39.928 99.8409 39.8295C100.129 39.7273 100.377 39.5758 100.585 39.375C100.794 39.1705 100.955 38.9167 101.068 38.6136L102.364 38.9773C102.227 39.4167 101.998 39.803 101.676 40.1364C101.354 40.4659 100.956 40.7235 100.483 40.9091C100.009 41.0909 99.4773 41.1818 98.8864 41.1818ZM105.305 32.2727L107.396 35.8409L109.487 32.2727H111.033L108.214 36.6364L111.033 41H109.487L107.396 37.6136L105.305 41H103.76L106.533 36.6364L103.76 32.2727H105.305ZM116.741 32.2727V33.4091H112.219V32.2727H116.741ZM113.537 30.1818H114.878V38.5C114.878 38.8788 114.933 39.1629 115.043 39.3523C115.156 39.5379 115.3 39.6629 115.474 39.7273C115.652 39.7879 115.84 39.8182 116.037 39.8182C116.185 39.8182 116.306 39.8106 116.401 39.7955C116.495 39.7765 116.571 39.7614 116.628 39.75L116.901 40.9545C116.81 40.9886 116.683 41.0227 116.52 41.0568C116.357 41.0947 116.151 41.1136 115.901 41.1136C115.522 41.1136 115.151 41.0322 114.787 40.8693C114.427 40.7064 114.128 40.4583 113.889 40.125C113.654 39.7917 113.537 39.3712 113.537 38.8636V30.1818ZM119.128 41V29.3636H120.537V39.75H125.946V41H119.128ZM127.946 41V32.2727H129.287V41H127.946ZM128.628 30.8182C128.366 30.8182 128.141 30.7292 127.952 30.5511C127.766 30.3731 127.673 30.1591 127.673 29.9091C127.673 29.6591 127.766 29.4451 127.952 29.267C128.141 29.089 128.366 29 128.628 29C128.889 29 129.113 29.089 129.298 29.267C129.488 29.4451 129.582 29.6591 129.582 29.9091C129.582 30.1591 129.488 30.3731 129.298 30.5511C129.113 30.7292 128.889 30.8182 128.628 30.8182ZM137.925 34.2273L136.72 34.5682C136.644 34.3674 136.533 34.1723 136.385 33.983C136.241 33.7898 136.044 33.6307 135.794 33.5057C135.544 33.3807 135.224 33.3182 134.834 33.3182C134.3 33.3182 133.855 33.4413 133.499 33.6875C133.146 33.9299 132.97 34.2386 132.97 34.6136C132.97 34.947 133.091 35.2102 133.334 35.4034C133.576 35.5966 133.955 35.7576 134.47 35.8864L135.766 36.2045C136.546 36.3939 137.127 36.6837 137.51 37.0739C137.893 37.4602 138.084 37.9583 138.084 38.5682C138.084 39.0682 137.94 39.5152 137.652 39.9091C137.368 40.303 136.97 40.6136 136.459 40.8409C135.947 41.0682 135.353 41.1818 134.675 41.1818C133.785 41.1818 133.048 40.9886 132.464 40.6023C131.881 40.2159 131.512 39.6515 131.357 38.9091L132.629 38.5909C132.75 39.0606 132.98 39.4129 133.317 39.6477C133.658 39.8826 134.103 40 134.652 40C135.277 40 135.773 39.8674 136.141 39.6023C136.512 39.3333 136.697 39.0114 136.697 38.6364C136.697 38.3333 136.591 38.0795 136.379 37.875C136.167 37.6667 135.841 37.5114 135.402 37.4091L133.947 37.0682C133.148 36.8788 132.561 36.5852 132.186 36.1875C131.815 35.786 131.629 35.2841 131.629 34.6818C131.629 34.1894 131.768 33.7538 132.044 33.375C132.324 32.9962 132.705 32.6989 133.186 32.483C133.671 32.267 134.22 32.1591 134.834 32.1591C135.697 32.1591 136.375 32.3485 136.868 32.7273C137.364 33.1061 137.716 33.6061 137.925 34.2273ZM143.898 32.2727V33.4091H139.375V32.2727H143.898ZM140.693 30.1818H142.034V38.5C142.034 38.8788 142.089 39.1629 142.199 39.3523C142.313 39.5379 142.456 39.6629 142.631 39.7273C142.809 39.7879 142.996 39.8182 143.193 39.8182C143.341 39.8182 143.462 39.8106 143.557 39.7955C143.652 39.7765 143.727 39.7614 143.784 39.75L144.057 40.9545C143.966 40.9886 143.839 41.0227 143.676 41.0568C143.513 41.0947 143.307 41.1136 143.057 41.1136C142.678 41.1136 142.307 41.0322 141.943 40.8693C141.583 40.7064 141.284 40.4583 141.045 40.125C140.811 39.7917 140.693 39.3712 140.693 38.8636V30.1818ZM73 62.8636C72.5417 62.8636 72.1231 62.7519 71.7443 62.5284C71.3655 62.3011 71.0625 61.9981 70.8352 61.6193C70.6117 61.2405 70.5 60.822 70.5 60.3636C70.5 59.9015 70.6117 59.483 70.8352 59.108C71.0625 58.7292 71.3655 58.428 71.7443 58.2045C72.1231 57.9773 72.5417 57.8636 73 57.8636C73.4621 57.8636 73.8807 57.9773 74.2557 58.2045C74.6345 58.428 74.9356 58.7292 75.1591 59.108C75.3864 59.483 75.5 59.9015 75.5 60.3636C75.5 60.822 75.3864 61.2405 75.1591 61.6193C74.9356 61.9981 74.6345 62.3011 74.2557 62.5284C73.8807 62.7519 73.4621 62.8636 73 62.8636ZM85.7727 54.6136V53.3636H94.5V54.6136H90.8409V65H89.4318V54.6136H85.7727ZM98.8864 65.1818C98.0455 65.1818 97.3201 64.9962 96.7102 64.625C96.1042 64.25 95.6364 63.7273 95.3068 63.0568C94.9811 62.3826 94.8182 61.5985 94.8182 60.7045C94.8182 59.8106 94.9811 59.0227 95.3068 58.3409C95.6364 57.6553 96.0947 57.1212 96.6818 56.7386C97.2727 56.3523 97.9621 56.1591 98.75 56.1591C99.2045 56.1591 99.6534 56.2348 100.097 56.3864C100.54 56.5379 100.943 56.7841 101.307 57.125C101.67 57.4621 101.96 57.9091 102.176 58.4659C102.392 59.0227 102.5 59.7083 102.5 60.5227V61.0909H95.7727V59.9318H101.136C101.136 59.4394 101.038 59 100.841 58.6136C100.648 58.2273 100.371 57.9223 100.011 57.6989C99.6553 57.4754 99.2348 57.3636 98.75 57.3636C98.2159 57.3636 97.7538 57.4962 97.3636 57.7614C96.9773 58.0227 96.6799 58.3636 96.4716 58.7841C96.2633 59.2045 96.1591 59.6553 96.1591 60.1364V60.9091C96.1591 61.5682 96.2727 62.1269 96.5 62.5852C96.7311 63.0398 97.0511 63.3864 97.4602 63.625C97.8693 63.8598 98.3447 63.9773 98.8864 63.9773C99.2386 63.9773 99.5568 63.928 99.8409 63.8295C100.129 63.7273 100.377 63.5758 100.585 63.375C100.794 63.1705 100.955 62.9167 101.068 62.6136L102.364 62.9773C102.227 63.4167 101.998 63.803 101.676 64.1364C101.354 64.4659 100.956 64.7235 100.483 64.9091C100.009 65.0909 99.4773 65.1818 98.8864 65.1818ZM105.305 56.2727L107.396 59.8409L109.487 56.2727H111.033L108.214 60.6364L111.033 65H109.487L107.396 61.6136L105.305 65H103.76L106.533 60.6364L103.76 56.2727H105.305ZM116.741 56.2727V57.4091H112.219V56.2727H116.741ZM113.537 54.1818H114.878V62.5C114.878 62.8788 114.933 63.1629 115.043 63.3523C115.156 63.5379 115.3 63.6629 115.474 63.7273C115.652 63.7879 115.84 63.8182 116.037 63.8182C116.185 63.8182 116.306 63.8106 116.401 63.7955C116.495 63.7765 116.571 63.7614 116.628 63.75L116.901 64.9545C116.81 64.9886 116.683 65.0227 116.52 65.0568C116.357 65.0947 116.151 65.1136 115.901 65.1136C115.522 65.1136 115.151 65.0322 114.787 64.8693C114.427 64.7064 114.128 64.4583 113.889 64.125C113.654 63.7917 113.537 63.3712 113.537 62.8636V54.1818ZM119.128 65V53.3636H120.537V63.75H125.946V65H119.128ZM127.946 65V56.2727H129.287V65H127.946ZM128.628 54.8182C128.366 54.8182 128.141 54.7292 127.952 54.5511C127.766 54.3731 127.673 54.1591 127.673 53.9091C127.673 53.6591 127.766 53.4451 127.952 53.267C128.141 53.089 128.366 53 128.628 53C128.889 53 129.113 53.089 129.298 53.267C129.488 53.4451 129.582 53.6591 129.582 53.9091C129.582 54.1591 129.488 54.3731 129.298 54.5511C129.113 54.7292 128.889 54.8182 128.628 54.8182ZM137.925 58.2273L136.72 58.5682C136.644 58.3674 136.533 58.1723 136.385 57.983C136.241 57.7898 136.044 57.6307 135.794 57.5057C135.544 57.3807 135.224 57.3182 134.834 57.3182C134.3 57.3182 133.855 57.4413 133.499 57.6875C133.146 57.9299 132.97 58.2386 132.97 58.6136C132.97 58.947 133.091 59.2102 133.334 59.4034C133.576 59.5966 133.955 59.7576 134.47 59.8864L135.766 60.2045C136.546 60.3939 137.127 60.6837 137.51 61.0739C137.893 61.4602 138.084 61.9583 138.084 62.5682C138.084 63.0682 137.94 63.5152 137.652 63.9091C137.368 64.303 136.97 64.6136 136.459 64.8409C135.947 65.0682 135.353 65.1818 134.675 65.1818C133.785 65.1818 133.048 64.9886 132.464 64.6023C131.881 64.2159 131.512 63.6515 131.357 62.9091L132.629 62.5909C132.75 63.0606 132.98 63.4129 133.317 63.6477C133.658 63.8826 134.103 64 134.652 64C135.277 64 135.773 63.8674 136.141 63.6023C136.512 63.3333 136.697 63.0114 136.697 62.6364C136.697 62.3333 136.591 62.0795 136.379 61.875C136.167 61.6667 135.841 61.5114 135.402 61.4091L133.947 61.0682C133.148 60.8788 132.561 60.5852 132.186 60.1875C131.815 59.786 131.629 59.2841 131.629 58.6818C131.629 58.1894 131.768 57.7538 132.044 57.375C132.324 56.9962 132.705 56.6989 133.186 56.483C133.671 56.267 134.22 56.1591 134.834 56.1591C135.697 56.1591 136.375 56.3485 136.868 56.7273C137.364 57.1061 137.716 57.6061 137.925 58.2273ZM143.898 56.2727V57.4091H139.375V56.2727H143.898ZM140.693 54.1818H142.034V62.5C142.034 62.8788 142.089 63.1629 142.199 63.3523C142.313 63.5379 142.456 63.6629 142.631 63.7273C142.809 63.7879 142.996 63.8182 143.193 63.8182C143.341 63.8182 143.462 63.8106 143.557 63.7955C143.652 63.7765 143.727 63.7614 143.784 63.75L144.057 64.9545C143.966 64.9886 143.839 65.0227 143.676 65.0568C143.513 65.0947 143.307 65.1136 143.057 65.1136C142.678 65.1136 142.307 65.0322 141.943 64.8693C141.583 64.7064 141.284 64.4583 141.045 64.125C140.811 63.7917 140.693 63.3712 140.693 62.8636V54.1818ZM73 86.8636C72.5417 86.8636 72.1231 86.7519 71.7443 86.5284C71.3655 86.3011 71.0625 85.9981 70.8352 85.6193C70.6117 85.2405 70.5 84.822 70.5 84.3636C70.5 83.9015 70.6117 83.483 70.8352 83.108C71.0625 82.7292 71.3655 82.428 71.7443 82.2045C72.1231 81.9773 72.5417 81.8636 73 81.8636C73.4621 81.8636 73.8807 81.9773 74.2557 82.2045C74.6345 82.428 74.9356 82.7292 75.1591 83.108C75.3864 83.483 75.5 83.9015 75.5 84.3636C75.5 84.822 75.3864 85.2405 75.1591 85.6193C74.9356 85.9981 74.6345 86.3011 74.2557 86.5284C73.8807 86.7519 73.4621 86.8636 73 86.8636ZM85.7727 78.6136V77.3636H94.5V78.6136H90.8409V89H89.4318V78.6136H85.7727ZM98.8864 89.1818C98.0455 89.1818 97.3201 88.9962 96.7102 88.625C96.1042 88.25 95.6364 87.7273 95.3068 87.0568C94.9811 86.3826 94.8182 85.5985 94.8182 84.7045C94.8182 83.8106 94.9811 83.0227 95.3068 82.3409C95.6364 81.6553 96.0947 81.1212 96.6818 80.7386C97.2727 80.3523 97.9621 80.1591 98.75 80.1591C99.2045 80.1591 99.6534 80.2348 100.097 80.3864C100.54 80.5379 100.943 80.7841 101.307 81.125C101.67 81.4621 101.96 81.9091 102.176 82.4659C102.392 83.0227 102.5 83.7083 102.5 84.5227V85.0909H95.7727V83.9318H101.136C101.136 83.4394 101.038 83 100.841 82.6136C100.648 82.2273 100.371 81.9223 100.011 81.6989C99.6553 81.4754 99.2348 81.3636 98.75 81.3636C98.2159 81.3636 97.7538 81.4962 97.3636 81.7614C96.9773 82.0227 96.6799 82.3636 96.4716 82.7841C96.2633 83.2045 96.1591 83.6553 96.1591 84.1364V84.9091C96.1591 85.5682 96.2727 86.1269 96.5 86.5852C96.7311 87.0398 97.0511 87.3864 97.4602 87.625C97.8693 87.8598 98.3447 87.9773 98.8864 87.9773C99.2386 87.9773 99.5568 87.928 99.8409 87.8295C100.129 87.7273 100.377 87.5758 100.585 87.375C100.794 87.1705 100.955 86.9167 101.068 86.6136L102.364 86.9773C102.227 87.4167 101.998 87.803 101.676 88.1364C101.354 88.4659 100.956 88.7235 100.483 88.9091C100.009 89.0909 99.4773 89.1818 98.8864 89.1818ZM105.305 80.2727L107.396 83.8409L109.487 80.2727H111.033L108.214 84.6364L111.033 89H109.487L107.396 85.6136L105.305 89H103.76L106.533 84.6364L103.76 80.2727H105.305ZM116.741 80.2727V81.4091H112.219V80.2727H116.741ZM113.537 78.1818H114.878V86.5C114.878 86.8788 114.933 87.1629 115.043 87.3523C115.156 87.5379 115.3 87.6629 115.474 87.7273C115.652 87.7879 115.84 87.8182 116.037 87.8182C116.185 87.8182 116.306 87.8106 116.401 87.7955C116.495 87.7765 116.571 87.7614 116.628 87.75L116.901 88.9545C116.81 88.9886 116.683 89.0227 116.52 89.0568C116.357 89.0947 116.151 89.1136 115.901 89.1136C115.522 89.1136 115.151 89.0322 114.787 88.8693C114.427 88.7064 114.128 88.4583 113.889 88.125C113.654 87.7917 113.537 87.3712 113.537 86.8636V78.1818ZM119.128 89V77.3636H120.537V87.75H125.946V89H119.128ZM127.946 89V80.2727H129.287V89H127.946ZM128.628 78.8182C128.366 78.8182 128.141 78.7292 127.952 78.5511C127.766 78.3731 127.673 78.1591 127.673 77.9091C127.673 77.6591 127.766 77.4451 127.952 77.267C128.141 77.089 128.366 77 128.628 77C128.889 77 129.113 77.089 129.298 77.267C129.488 77.4451 129.582 77.6591 129.582 77.9091C129.582 78.1591 129.488 78.3731 129.298 78.5511C129.113 78.7292 128.889 78.8182 128.628 78.8182ZM137.925 82.2273L136.72 82.5682C136.644 82.3674 136.533 82.1723 136.385 81.983C136.241 81.7898 136.044 81.6307 135.794 81.5057C135.544 81.3807 135.224 81.3182 134.834 81.3182C134.3 81.3182 133.855 81.4413 133.499 81.6875C133.146 81.9299 132.97 82.2386 132.97 82.6136C132.97 82.947 133.091 83.2102 133.334 83.4034C133.576 83.5966 133.955 83.7576 134.47 83.8864L135.766 84.2045C136.546 84.3939 137.127 84.6837 137.51 85.0739C137.893 85.4602 138.084 85.9583 138.084 86.5682C138.084 87.0682 137.94 87.5152 137.652 87.9091C137.368 88.303 136.97 88.6136 136.459 88.8409C135.947 89.0682 135.353 89.1818 134.675 89.1818C133.785 89.1818 133.048 88.9886 132.464 88.6023C131.881 88.2159 131.512 87.6515 131.357 86.9091L132.629 86.5909C132.75 87.0606 132.98 87.4129 133.317 87.6477C133.658 87.8826 134.103 88 134.652 88C135.277 88 135.773 87.8674 136.141 87.6023C136.512 87.3333 136.697 87.0114 136.697 86.6364C136.697 86.3333 136.591 86.0795 136.379 85.875C136.167 85.6667 135.841 85.5114 135.402 85.4091L133.947 85.0682C133.148 84.8788 132.561 84.5852 132.186 84.1875C131.815 83.786 131.629 83.2841 131.629 82.6818C131.629 82.1894 131.768 81.7538 132.044 81.375C132.324 80.9962 132.705 80.6989 133.186 80.483C133.671 80.267 134.22 80.1591 134.834 80.1591C135.697 80.1591 136.375 80.3485 136.868 80.7273C137.364 81.1061 137.716 81.6061 137.925 82.2273ZM143.898 80.2727V81.4091H139.375V80.2727H143.898ZM140.693 78.1818H142.034V86.5C142.034 86.8788 142.089 87.1629 142.199 87.3523C142.313 87.5379 142.456 87.6629 142.631 87.7273C142.809 87.7879 142.996 87.8182 143.193 87.8182C143.341 87.8182 143.462 87.8106 143.557 87.7955C143.652 87.7765 143.727 87.7614 143.784 87.75L144.057 88.9545C143.966 88.9886 143.839 89.0227 143.676 89.0568C143.513 89.0947 143.307 89.1136 143.057 89.1136C142.678 89.1136 142.307 89.0322 141.943 88.8693C141.583 88.7064 141.284 88.4583 141.045 88.125C140.811 87.7917 140.693 87.3712 140.693 86.8636V78.1818Z" fill="#233D48"/>
</svg>
