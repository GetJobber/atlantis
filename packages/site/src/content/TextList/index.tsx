import Content from "@atlantis/docs/components/TextList/TextList.stories.mdx";
import MobileProps from "./TextList.props-mobile.json";
import { ContentExport } from "../../types/content";

export default {
  content: () => <Content />,
  mobileProps: MobileProps,
  component: {
    mobileElement: `<TextList items={["Item uno", "Item dos", "Item tres"]} />`,
  },
  title: "TextList",
  links: [
    {
      label: "Storybook",
      url: `${
        (window as AtlantisWindow)?.env?.VITE_STORYBOOK_URL
      }?path=/docs/components-lists-and-tables-textlist--docs`,
    },
  ],
} as const satisfies ContentExport;
