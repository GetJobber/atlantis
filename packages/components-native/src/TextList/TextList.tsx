import React from "react";
import { View } from "react-native";
import { useStyles } from "./TextList.style";
import type { Spacing } from "../Content";
import { Content } from "../Content";
import type { TextLevel } from "../Text";
import { Text } from "../Text";

const BULLET_SYMBOL = `\u2022`;

interface TextListProps {
  /**
   * Text to display.
   */
  readonly items?: string[];

  /**
   * Change the appearance of the text
   */
  readonly emphasis?: "strong";

  /**
   * Visual hierarchy of the text.
   * @default "text"
   */
  readonly level?: TextLevel;

  /**
   * The amount of spacing that content will give.
   * @default "none"
   */
  readonly spacing?: Spacing;

  /**
   * The amount of spacing that will be applied between the list items.
   * @default "none"
   */
  readonly childSpacing?: Spacing;
}

export function TextList({
  items,
  childSpacing = "none",
  emphasis,
  level = "text",
  spacing = "none",
}: TextListProps): JSX.Element {
  const styles = useStyles();

  return (
    <>
      {items && (
        <View style={styles.details}>
          <Content spacing={spacing} childSpacing={childSpacing}>
            {items.map((item, index) => (
              <View style={styles.detail} key={index}>
                <Text level={level} emphasis={emphasis}>
                  {BULLET_SYMBOL}
                </Text>
                <View style={styles.detailText}>
                  <Text level={level} emphasis={emphasis}>
                    {item}
                  </Text>
                </View>
              </View>
            ))}
          </Content>
        </View>
      )}
    </>
  );
}
