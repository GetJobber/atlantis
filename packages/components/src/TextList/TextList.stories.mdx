import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
import { TextList } from "@jobber/components/TextList";

<Meta title="Components/TextList" component={TextList} />

# TextList

```ts
import { TextList } from "@jobber/components/TextList";
```

<Canvas>
  <Story
    name="TextList"
    args={{
      data: [
        "Your name",
        "Company name (if applicable)",
        "Phone number",
        "Email address",
        "Property address",
      ],
    }}
  >
    {args => <TextList {...args} />}
  </Story>
</Canvas>

<ArgsTable of={TextList} story="TextList" />
