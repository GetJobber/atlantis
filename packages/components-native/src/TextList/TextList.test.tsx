import React from "react";
import { render } from "@testing-library/react-native";
import { TextList } from "./TextList";

describe("TextList", () => {
  describe("when the bulletItems is provided", () => {
    it("displays the text list", () => {
      const items = ["this is list item uno", "this is list item dos"];
      const tree = render(<TextList items={items} />);

      expect(tree.toJSON()).toMatchSnapshot();
    });

    it("displays will not display a bulleted list", () => {
      const tree = render(<TextList />);

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
