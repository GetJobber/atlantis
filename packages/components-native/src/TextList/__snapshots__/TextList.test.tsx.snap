// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextList when the bulletItems is provided displays the text list 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {},
        {
          "padding": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {},
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessibilityRole="text"
          adjustsFontSizeToFit={false}
          allowFontScaling={true}
          collapsable={false}
          maxFontSizeMultiplier={3.125}
          selectable={true}
          selectionColor="hsl(86, 100%, 46%)"
          style={
            [
              {
                "fontFamily": "inter-regular",
              },
              {
                "color": "hsl(198, 35%, 21%)",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 16,
                "lineHeight": 20,
              },
              {
                "letterSpacing": 0,
              },
            ]
          }
        >
          •
        </Text>
        <View
          style={
            {
              "flex": 1,
              "paddingLeft": 8,
            }
          }
        >
          <Text
            accessibilityRole="text"
            adjustsFontSizeToFit={false}
            allowFontScaling={true}
            collapsable={false}
            maxFontSizeMultiplier={3.125}
            selectable={true}
            selectionColor="hsl(86, 100%, 46%)"
            style={
              [
                {
                  "fontFamily": "inter-regular",
                },
                {
                  "color": "hsl(198, 35%, 21%)",
                },
                {
                  "textAlign": "left",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 20,
                },
                {
                  "letterSpacing": 0,
                },
              ]
            }
          >
            this is list item uno
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {},
          {
            "padding": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessibilityRole="text"
          adjustsFontSizeToFit={false}
          allowFontScaling={true}
          collapsable={false}
          maxFontSizeMultiplier={3.125}
          selectable={true}
          selectionColor="hsl(86, 100%, 46%)"
          style={
            [
              {
                "fontFamily": "inter-regular",
              },
              {
                "color": "hsl(198, 35%, 21%)",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 16,
                "lineHeight": 20,
              },
              {
                "letterSpacing": 0,
              },
            ]
          }
        >
          •
        </Text>
        <View
          style={
            {
              "flex": 1,
              "paddingLeft": 8,
            }
          }
        >
          <Text
            accessibilityRole="text"
            adjustsFontSizeToFit={false}
            allowFontScaling={true}
            collapsable={false}
            maxFontSizeMultiplier={3.125}
            selectable={true}
            selectionColor="hsl(86, 100%, 46%)"
            style={
              [
                {
                  "fontFamily": "inter-regular",
                },
                {
                  "color": "hsl(198, 35%, 21%)",
                },
                {
                  "textAlign": "left",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 20,
                },
                {
                  "letterSpacing": 0,
                },
              ]
            }
          >
            this is list item dos
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`TextList when the bulletItems is provided displays will not display a bulleted list 1`] = `null`;
